src/docs/arch-cursor.dox

```diff
@@ -371,6 +371,15 @@ how many references to closed data handles are freed, we may continue our walk.
 usually strike a good balance between not having a lot of overhead for sweeps, and keeping up with
 the need to free up shared resources.
 
+@section arch_cursor_bound Range bounded cursors
+
+The lower and upper bound in a cursor is represented as a \c WT_ITEM. The implementation applies bounds through the function
+__wt_cursor_bound which takes the cursor and a config string as arguments. The config string specifies whether the operation is clearing
+or setting the bound, and if a "set" is in progress, whether the bound is a lower or upper bound. There are four flags that can get set
+when applying bounds: WT_CURSTD_BOUND_LOWER, WT_CURSTD_BOUND_LOWER_INCLUSIVE, WT_CURSTD_BOUND_UPPER, and WT_CURSTD_BOUND_UPPER_INCLUSIVE.
+When calling __wt_cursor_bound with valid bounds, the respective bound flag gets set and the bound key data saved in the cursor.
+Clearing a cursor's bounds can be done via resetting the cursor or calling bound with the "clear" config which will free both bounds buffers.
```

Comment:
I realized we should include a bit more detail here about the performance optimization we get from bounded cursors. 

We should mention that the bounds checking ignores whether the value associated with the key is visible to the cursor and can return `WT_NOTFOUND` when it detects a key that is outside the range.

Reply:
I have included mention of the performance optimization we get here [6b88eb1](https://github.com/wiredtiger/wiredtiger/pull/8367/commits/6b88eb16841dd50de96ea1436b961ec3a0e8644d). I did not go very far into detail with it, though, so let me know if I should have written more.

Reply:
I would rephrase the first few sentences  a bit:
```
Range bounded cursors provide a mechanism for optimizing cursor operations in collections with a large number of invisible records. When traversing records the record's key is compared with the bounds set on the cursor and the traversal will exit with WT_NOTFOUND regardless of whether the value associated with the cursor is visible.
```